src/main/kotlin/org/partiql/scribe/ScribeCompiler.kt

```diff
@@ -55,9 +55,18 @@ public class ScribeCompiler internal constructor(
         return result.root
     }
 
+    private class PlanProblemCallback: ProblemCallback {
+        val problems = mutableListOf<Problem>()
+
+        override fun invoke(p1: Problem) {
+            problems.add(p1)
+        }
+    }
+
     private fun plan(statement: Statement, session: PartiQLPlanner.Session): PartiQLPlan {
-        val result = planner.plan(statement, session)
-        val errors = result.problems.filter { it.details.severity == ProblemSeverity.ERROR }
```

Comment:
`PartiQLPlanner` result is currently a confusing API (see https://github.com/partiql/partiql-lang-kotlin/issues/1396). `PartiQLPlanner.problems` would always have an empty list of problems